src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateModified = [];
+  const stateObject = { ...state };
+
+  for (const obj of actions) {
+    for (const key in obj) {
```

Comment:
Nested loop is redundant here, also it's a bad practice for performance.
Consider using only one here